### DISCUSSION TITLE
3-way task-list items with Unchecked, Checked and Not-Applicable

### DISCUSSION THREAD

**Discussion by markcmiller86** [community/community#2922] [Discussions]
Comments in PRs and issues on GitHub support task lists (like the one pictured below).

- [ ] one task
   - [ ] sub task 1
   - [ ] sub task 2
- [ ] another task
- [ ] yet another task

The items can either be checked or UN-checked. I am interested in gaining support for an enhancement to support a 3rd designation for *Not Applicable* similar, somewhat, to you see pictured [in this hack](https://github.com/visit-dav/visit/issues/5295).

We use these in PR templates to a) remind contributors of the things they should include in their PR and b) to indicate whether a PR has all completed items. However, not all PRs need all the check-list items we include in our templates. So, we'd like this 3rd designation to allow contributors to indicate they think that item is not applicable in their PR.

Clicking could cycle through 3 options (UNchecked, checked, N/A) or maybe a shift-click toggles between unchecked and N/A.

Also, when reporting PR's status where the number of tasks completed is displayed, only the tasks that are NOT designated as N/A should be counted.

**Comment by becca** [comment]
Thank you for this feedback @markcmiller86! Are you suggesting that this be supported only in Discussions or also across PRs and Issues (I assume all of the above!)

cc// @labrose

**Comment by markcmiller86** [comment]
Yes, all of the above. Though, priority would be PRs because that is where a we use templates that can include not-applicable items and it is also where *status* is displayed in the PR headline showing something like "6 of 9 tasks completed"

**Reply by labrose** [reply]
Love your hacks above, I use the bullet one myself 😄 

What's the biggest drawback you're encountering from the workarounds? Just want to make sure I understand your request and use case.

**Reply by markcmiller86** [reply]
> What's the biggest drawback you're encountering from the workarounds?

Non obvious how to designate `not-applicable` to non-cognizant users. More typing and mouse clicks. You have to go into *Edit* mode to change something to `not-applicable`.

**Reply by chadwhitacre** [reply]
I use check + strikethrough to indicate that an item was resolved by dropping it from scope.

- [x] ~something that was dropped from scope~

**Reply by markcmiller86** [reply]
I do something similar (example [here](https://github.com/visit-dav/visit/pull/17599)) but include the checkbox *inside* the struck-out line. That too winds up causing it to NOT show in the total count of items to be completed.
